src/Microsoft.Health.Fhir.Shared.Core/Features/Resources/Delete/DeletionService.cs

```diff
@@ -51,20 +55,22 @@ public class DeletionService : IDeletionService
         public DeletionService(
             IResourceWrapperFactory resourceWrapperFactory,
             Lazy<IConformanceProvider> conformanceProvider,
-            IFhirDataStore fhirDataStore,
-            ISearchService searchService,
+            Func<IScoped<IFhirDataStore>> fhirDataStoreFactory,
+            Func<IScoped<ISearchService>> searchServiceFactory,
```

Comment:
This should work to resolve a new instance, but every invoke leaves a floating IScope<> object reference in the root container

Reply:
I'm working to review your PR so that it shouldn't be an issue anymore.